go/pkg/tool/embeddedtool.go

```diff
@@ -157,3 +159,13 @@ func getPathFlag() string {
 	}
 	return pathPrefix
 }
+
+func validateActionRootAndDg() (string, string) {
```

Comment:
nit: `validate` implies no return values, but validation can be expected as a side effect for a function that does return values.
```suggestion
func inputDgAndActionRoot() (string, string) {
```